src/services/symbolDisplay.ts

```diff
@@ -489,8 +489,10 @@ namespace ts.SymbolDisplay {
             addNewLineIfDisplayPartsExist();
             if (symbolKind) {
                 pushTypePart(symbolKind);
-                displayParts.push(spacePart());
-                addFullSymbolName(symbol);
+                if (!some(symbol.declarations, d => isArrowFunction(d) || (isFunctionExpression(d) || isClassExpression(d)) && !d.name)) {

```

Comment:
`(d => !d.name)` ?

Reply:
A few declarations don't have a `name` property but will still have a name when written out by the checker.